internal/scorecard/helpers/helpers.go

```diff
@@ -132,47 +54,69 @@ func TestResultToScorecardTestResult(tr TestResult) scapiv1alpha1.ScorecardTestR
+	var failFound bool
+	finalResult := TestResult{
+		State: scapiv1alpha2.PassState,
 	}
-
-	if res.EarnedPoints == 0 {
-		res.State = scapiv1alpha1.FailState
-	} else if res.EarnedPoints < res.MaximumPoints {
-		res.State = scapiv1alpha1.PartialPassState
-	} else if res.EarnedPoints == res.MaximumPoints {
-		res.State = scapiv1alpha1.PassState
+	if len(results) > 0 {
+		name = results[0].Test.GetName()
+		// all results have the same test
+		finalResult.Test = results[0].Test
+	}
+	for _, result := range results {
+		if !result.Test.IsCumulative() {
+			return finalResult, fmt.Errorf("non-cumulative test passed to ResultsCumulative: name (%s)", result.Test.GetName())
+		}
+		if result.Test.GetName() != name {
+			return finalResult, fmt.Errorf("test name mismatch in ResultsCumulative: %s != %s", result.Test.GetName(), name)
+		}
+		finalResult.Suggestions = append(finalResult.Suggestions, result.Suggestions...)
+		finalResult.Errors = append(finalResult.Errors, result.Errors...)
+		if result.State != scapiv1alpha2.PassState {
+			failFound = true
+		}
+	}
+	if failFound {
+		finalResult.State = scapiv1alpha2.FailState
 	}
-	return res
-	// TODO: decide what to do if a Test incorrectly sets points (Earned > Max)
+	return finalResult, nil
 }
 
-// UpdateSuiteStates update the state of each test in a suite and updates the count to the suite's states to match
-func UpdateSuiteStates(suite scapiv1alpha1.ScorecardSuiteResult) scapiv1alpha1.ScorecardSuiteResult {
-	totalTests := len(suite.Tests)
-	if totalTests == 0 {
-		return suite
+// ResultsPassFail combines multiple test results and returns a
+// single test result
+func ResultsPassFail(results []TestResult) (TestResult, error) {
+	var name string
+	var failFound bool
+	finalResult := TestResult{
+		State: scapiv1alpha2.PassState,
+	}
+	if len(results) > 0 {
+		name = results[0].Test.GetName()
+		// all results have the same test
+		finalResult.Test = results[0].Test
 	}
-	suite.TotalTests = totalTests
-	// reset all state values
-	suite.Error = 0
-	suite.Fail = 0
-	suite.PartialPass = 0
-	suite.Pass = 0
-	for idx, test := range suite.Tests {
-		suite.Tests[idx] = UpdateState(test)
-		switch test.State {
-		case scapiv1alpha1.ErrorState:
-			suite.Error++
-		case scapiv1alpha1.PassState:
-			suite.Pass++
-		case scapiv1alpha1.PartialPassState:
-			suite.PartialPass++
-		case scapiv1alpha1.FailState:
-			suite.Fail++
+	for _, result := range results {
+		if result.Test.IsCumulative() {
+			return finalResult, fmt.Errorf("cumulative test passed to ResultsPassFail: name (%s)", result.Test.GetName())
+		}
+		if result.Test.GetName() != name {
+			return finalResult, fmt.Errorf("test name mismatch in ResultsPassFail: %s != %s", result.Test.GetName(), name)
+		}
+		finalResult.Suggestions = append(finalResult.Suggestions, result.Suggestions...)
+		finalResult.Errors = append(finalResult.Errors, result.Errors...)
+		finalResult.Log = result.Log
+		if result.State == scapiv1alpha2.FailState {
```

Comment:
Should this match what we're doing in `ResultsCumulative()`?

Also, what about the case where one of the results has `scapiv1alpha2.ErrorState`?

Reply:
I updated that to match, also found a bug in exit code for the list flag while making the update.  Good catch!